Regression (11.19.2): a root tsconfig with `references` drops its own `paths` for files matched by the default `include` (`**/*`)

## Summary

Since 11.19.2, a tsconfig that has `references` and **omits** `files`/`include` no longer applies its own `compilerOptions.paths` to the files it owns through the default `**/*` include. Such configs are now treated as "solution-style, owns nothing", but per TypeScript an **omitted** `include` defaults to `**/*` — only an **explicit** `files: []` / `include: []` means "own no files".

This regressed the area around #1075 via the project-reference ownership changes in #1151 / #1154.

## Reproduction

```
repro/
├── tsconfig.json        { compilerOptions: { baseUrl: ".", paths: { "@app/*": ["./src/*"] } },
│                          references: [{ "path": "./pkg" }] }   ← no files / no include
├── index.ts             import { util } from '@app/util';
├── src/util.ts          export const util = () => 'util';
└── pkg/
    ├── tsconfig.json    { compilerOptions: { composite: true } }
    └── thing.ts
```

```js
// repro.mjs  —  npm i oxc-resolver && node repro.mjs
import { ResolverFactory } from 'oxc-resolver';
import { fileURLToPath } from 'node:url';

const dir = fileURLToPath(new URL('.', import.meta.url));
const resolver = new ResolverFactory({ tsconfig: 'auto', extensions: ['.ts', '.tsx', '.d.ts'] });
console.log(resolver.resolveFileSync(dir + 'index.ts', '@app/util'));
```

## Expected vs actual

`@app/util` resolved from `index.ts` (which the root tsconfig owns via the default `**/*`, and which **no** reference covers):

| | result |
|---|---|
| `tsc` 6.0.3 `--traceResolution` | ✅ `src/util.ts` |
| `tsgo` 7.0.0-dev `--traceResolution` | ✅ `src/util.ts` |
| oxc-resolver **11.19.1** | ✅ `src/util.ts` |
| oxc-resolver **11.19.2** | ❌ `Cannot find module '@app/util'` |

Both TypeScript implementations resolve it; 11.19.2 does not.

## Root cause

`src/tsconfig.rs`, `claims_ownership_of`:

```rust
let is_solution_style = !self.references_resolved.is_empty()
    && self.files.as_ref().is_none_or(Vec::is_empty)      // None (omitted) == Some([])
    && self.include.as_ref().is_none_or(Vec::is_empty);
if is_solution_style { return false; }
```

`is_none_or(Vec::is_empty)` collapses **omitted** and **explicitly empty** into one case. Per the [TS `include` default](https://www.typescriptlang.org/tsconfig/#include) — *"`[]` if `files` is specified; `**/*` otherwise"* — only an explicit empty array means "own nothing"; an omitted `include` should fall through to the existing `**/*` check (`is_file_included_in_tsconfig`). Suggested gate: `matches!(self.include.as_deref(), Some([]))` (and the same for `files`) instead of `is_none_or`.

This also matches the algorithm documented in `src/tsconfig_resolver.rs`: *"2.2. If none of the references include this path, return the `tsconfig.json` found in step 1."*

## Environment

- oxc-resolver 11.19.2 (built from tag `v11.19.2`); also reproduces on `main`. Works on 11.19.1.

--- a/fixtures/tsconfig/cases/project-references-default-include/index.ts
+++ b/fixtures/tsconfig/cases/project-references-default-include/index.ts
@@ -1,0 +1,3 @@
+import { util } from "@app/util";
+
+util();

--- a/fixtures/tsconfig/cases/project-references-default-include/pkg/thing.ts
+++ b/fixtures/tsconfig/cases/project-references-default-include/pkg/thing.ts
@@ -1,0 +1,1 @@
+export const thing = () => "thing";

--- a/fixtures/tsconfig/cases/project-references-default-include/pkg/tsconfig.json
+++ b/fixtures/tsconfig/cases/project-references-default-include/pkg/tsconfig.json
@@ -1,0 +1,5 @@
+{
+  "compilerOptions": {
+    "composite": true
+  }
+}

--- a/fixtures/tsconfig/cases/project-references-default-include/src/util.ts
+++ b/fixtures/tsconfig/cases/project-references-default-include/src/util.ts
@@ -1,0 +1,1 @@
+export const util = () => "util";

--- a/fixtures/tsconfig/cases/project-references-default-include/tsconfig.json
+++ b/fixtures/tsconfig/cases/project-references-default-include/tsconfig.json
@@ -1,0 +1,7 @@
+{
+  "compilerOptions": {
+    "baseUrl": ".",
+    "paths": { "@app/*": ["./src/*"] }
+  },
+  "references": [{ "path": "./pkg" }]
+}

--- a/src/tests/tsconfig_project_references.rs
+++ b/src/tests/tsconfig_project_references.rs
@@ -204,3 +204,18 @@ fn walk_up_when_ref_excludes_file() {
         resolver.resolve_file(f.join("pkg-a/src/excluded/bar.ts"), "@/foo").map(|f| f.full_path());
     assert_eq!(resolved_path, Err(ResolveError::NotFound("@/foo".into())));
 }
+
+#[test]
+fn root_paths_apply_to_default_include_files() {
+    let f = super::fixture_root().join("tsconfig/cases/project-references-default-include");
+
+    let resolver = Resolver::new(ResolveOptions {
+        extensions: vec![".ts".into()],
+        tsconfig: Some(TsconfigDiscovery::Auto),
+        ..ResolveOptions::default()
+    });
+
+    let resolved_path =
+        resolver.resolve_file(f.join("index.ts"), "@app/util").map(|f| f.full_path());
+    assert_eq!(resolved_path, Ok(f.join("src/util.ts")));
+}

--- a/src/tsconfig.rs
+++ b/src/tsconfig.rs
@@ -24,8 +24,6 @@ use crate::{TsconfigReferences, path::PathUtil, replace_bom_with_whitespace};
 /// Allow list: <https://github.com/microsoft/TypeScript/issues/57485#issuecomment-2027787456>
 const TEMPLATE_VARIABLE: &str = "${configDir}";
 
-const GLOB_ALL_PATTERN: &str = "**/*";
-
 pub type CompilerOptionsPathsMap = IndexMap<String, Vec<PathBuf>, BuildHasherDefault<FxHasher>>;
 
 /// Project Reference
@@ -578,11 +576,14 @@ impl TsConfig {
         if self.references_resolved.iter().any(|r| r.is_file_included_in_tsconfig(path)) {
             return true;
         }
-        // Solution-style configs (have `references` and no own `files` /
-        // `include`) never claim files themselves.
+        // Solution-style configs (have `references` and explicit empty
+        // `files` / `include`) never claim files themselves. Per the
+        // TypeScript spec, an *omitted* `include` defaults to `**/*`, so it
+        // must fall through to `is_file_included_in_tsconfig`; only an
+        // explicit empty array means "own no files".
         let is_solution_style = !self.references_resolved.is_empty()
-            && self.files.as_ref().is_none_or(Vec::is_empty)
-            && self.include.as_ref().is_none_or(Vec::is_empty);
+            && matches!(self.files.as_deref(), Some([]))
+            && matches!(self.include.as_deref(), Some([]));
         if is_solution_style {
             return false;
         }
@@ -618,7 +619,8 @@ impl TsConfig {
     fn is_glob_matches(&self, path: &Path, pattern: GlobPattern) -> bool {
         let path_str = path.to_string_lossy().replace('\\', "/");
         match pattern {
-            GlobPattern::All => self.is_glob_match(GLOB_ALL_PATTERN, path, &path_str),
+            // The default include `**/*` is scoped to the tsconfig directory.
+            GlobPattern::All => path.starts_with(self.directory()),
             GlobPattern::Pattern(patterns) => patterns.iter().any(|pattern| {
                 let pattern = pattern.to_string_lossy().replace('\\', "/");
                 self.is_glob_match(pattern.as_ref(), path, &path_str)
@@ -628,10 +630,6 @@ impl TsConfig {
 
     fn is_glob_match(&self, pattern: &str, path: &Path, path_str: &str) -> bool {
         if pattern == path_str {
-            return true;
-        }
-        // Special case: **/* matches everything
-        if pattern == GLOB_ALL_PATTERN {
             return true;
         }
         // Normalize pattern: add implicit /**/* for directory patterns

